### DISCUSSION TITLE
[Question]  extends WithEventBus 和  Autowired EventBus 应该使用哪种方式更加合理一些？

### DISCUSSION THREAD

**Discussion by zhuzeyu22** [opensumi/core#3291] [General]
extends WithEventBus 和  Autowired EventBus 都可以使用 事件机制，用哪种更加优雅一些？

**Comment by bytemain** [comment]
全凭对代码风格的喜好，`WithEventBus` 背后的实现还是 `Autowired EventBus`。

`extends WithEventBus` 后可以让你通过装饰器的方式来接收事件（`@OnEvent`）:
![CleanShot 2024-01-22 at 17 22 46@2x](https://github.com/opensumi/core/assets/13938334/6a184d0f-4aee-472d-95f0-b50d493ef329)

我个人认为上图这种的写法更简洁，代码风格更好。
